internal/handlers/health.go

```diff
@@ -2,10 +2,29 @@ package handlers
 
 import (
 	"net/http"
+
+	log "github.com/sirupsen/logrus"
 )
 
-func NewHealthHandler() http.Handler {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusOK)
-	})
+type HealthHandler struct {
+	isEnabled bool
+}
+
+func NewHealthHandler() *HealthHandler {
```

Comment:
Maybe name this after the probe it's implementing like you did with the other one?
So this would be a `ReadinessHandler` in that case.